src/commands/test/cheatcodes/syscall_handler.py

```diff
@@ -53,6 +56,44 @@ def _get_caller_address(
             return self.custom_caller_address
         return super()._get_caller_address(segments, syscall_ptr)
 
+    # mock_call
+    mocked_calls: Dict[AddressType, Dict[SelectorType, List[int]]] = {}
+
+    def register_mock_call(
+        self, contract_address: AddressType, selector: int, ret_data: List[int]
+    ):
+        if contract_address not in self.mocked_calls:
+            self.mocked_calls[contract_address] = {}
+        self.mocked_calls[contract_address][selector] = ret_data
+
+    def unregister_mock_call(self, contract_address: AddressType, selector: int):
+        if contract_address not in self.mocked_calls:
+            raise Exception(
+                f"Contract {contract_address} doesn't have mocked selectors."
+            )
+        if selector not in self.mocked_calls[contract_address]:
+            raise Exception(
```

Comment:
What do you think about making this exception custom

Reply:
I can create `CheatcodeException`  to avoid throwing generic exceptions.